graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/RemoteReindexingMigrationAdapterOS2.java

```diff
@@ -104,14 +108,17 @@ public void start(final URI uri, final String username, final String password, f
 
     @Override
     public Status status() {
+        // TODO: return real migration status
         return Status.STARTING;
     }
 
     Set<String> getActiveNodes() {
-        return nodeService.allActive().values().stream().map(NodeDto::getNodeId).collect(Collectors.toSet());
+        return nodeService.allActive().values().stream()
+                .filter(dn -> dn.getDataNodeStatus() == DataNodeStatus.AVAILABLE) // we have to wait till the datanode is not just alive but all started properly and the indexer accepts connections
```

Comment:
Here we need to wait till the status is AVAILABLE, otherwise we may get an datanode that's running but the underlying opensearch is still starting. Then any connection will be refused.